base_import_match/models/base_import.py

```diff
@@ -0,0 +1,297 @@
+# -*- coding: utf-8 -*-
+# © 2016 Grupo ESOC Ingeniería de Servicios, S.L.U. - Jairo Llopis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+from openerp import SUPERUSER_ID  # TODO remove in v10
+
+
+class BaseImportMatch(models.Model):
+    _name = "base_import.match"
+    _description = "Deduplicate settings prior to CSV imports."
+    _order = "sequence, name"
+
+    name = fields.Char(
+        compute="_compute_name",
+        store=True,
+        index=True)
+    sequence = fields.Integer(index=True)
+    model_id = fields.Many2one(
+        "ir.model",
+        "Model",
+        required=True,
+        ondelete="cascade",
+        domain=[("osv_memory", "=", False)],
+        help="In this model you will apply the match.")
+    model_name = fields.Char(
+        related="model_id.model",
+        store=True,
+        index=True)
+    field_ids = fields.One2many(
+        comodel_name="base_import.match.field",
+        inverse_name="match_id",
+        string="Fields",
+        required=True,
+        help="Fields that will define an unique key.")
+
+    @api.multi
+    @api.onchange("model_id")
+    def _onchange_model_id(self):
+        self.field_ids.unlink()
+
+    @api.model
+    def create(self, vals):
+        """Wrap the model after creation."""
+        result = super(BaseImportMatch, self).create(vals)
+        self._load_autopatch(result.model_name)
```

Comment:
This only patches the modules found in configuration. That's why that method is called when creating, deleting, writing and restarting Odoo: it patches configured models and unpatches those that not.